cookbook/basic_rag_example/pinecone_embeddings.py

```diff
@@ -0,0 +1,151 @@
+"""A script using RAG and Pinecone to summarize relevant articles from a large database.
+
+First we load in a dataset of news articles from 2004 and embed them. Then we use
+Pinecone to efficiently find the articles with most relevance to a query, then use said
+articles in a prompt to get the summary.
+"""
+import ast
+import os
+
+import pandas as pd
+import tiktoken
+from pinecone import Pinecone, ServerlessSpec
+
+from mirascope import OpenAIChat, Prompt
+
+os.environ["OPENAI_API_KEY"] = "YOUR_API_KEY"
+chat = OpenAIChat()
+embeddings_model = "text-embedding-ada-002"
+
+
+def load_data(url) -> pd.DataFrame:
+    """Loads data from a url, and splits larger texts into smaller chunks."""
+    df = pd.read_csv(url)
+    split_articles = []
+    encoder = tiktoken.encoding_for_model("gpt-3.5-turbo")
+
+    def split_text(text, tokens, max_tokens):
```

Comment:
move this outside of load_data as it's own function, add type hints to the parameters, and flesh out the docstring to be a full docstring